File: library/src/main/java/me/proxer/library/api/anime/InternalApi.java
package me.proxer.library.api.anime;

import me.proxer.library.api.ProxerCall;
import me.proxer.library.entitiy.anime.Stream;
import me.proxer.library.enums.AnimeLanguage;
import retrofit2.http.GET;
import retrofit2.http.Query;

import javax.annotation.ParametersAreNullableByDefault;
import java.util.List;

/**
 * @author Ruben Gees
 */
@ParametersAreNullableByDefault
interface InternalApi {

    @GET("anime/streams")
    ProxerCall<List<Stream>> streams(@Query("id") String id,
                                     @Query("episode") Integer episode,
                                     @Query("language") AnimeLanguage language);

    @GET("anime/proxerstreams")
    ProxerCall<List<Stream>> proxerStreams(@Query("id") String id,
                                           @Query("episode") Integer episode,
                                           @Query("language") AnimeLanguage language);

    @GET("anime/link")
    ProxerCall<String> link(@Query("id") String id);
}

File: library/src/main/kotlin/me/proxer/library/api/anime/InternalApi.kt
package me.proxer.library.api.anime

import me.proxer.library.ProxerCall
import me.proxer.library.entity.anime.LinkContainer
import me.proxer.library.entity.anime.Stream
import me.proxer.library.enums.AnimeLanguage
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * @author Ruben Gees
 */
internal interface InternalApi {

    @GET("anime/streams")
    fun streams(
        @Query("id") id: String?,
        @Query("episode") episode: Int?,
        @Query("language") language: AnimeLanguage?
    ): ProxerCall<List<Stream>>

    @GET("anime/proxerstreams")
    fun proxerStreams(
        @Query("id") id: String?,
        @Query("episode") episode: Int?,
        @Query("language") language: AnimeLanguage?
    ): ProxerCall<List<Stream>>

    @GET("anime/link")
    fun link(
        @Query("id") id: String?
    ): ProxerCall<String>

    @GET("anime/linkvast")
    fun vastLink(
        @Query("id") id: String?
    ): ProxerCall<LinkContainer>
}
